persistence/relational-jdbc/src/main/java/org/apache/polaris/persistence/relational/jdbc/RelationalJdbcConfiguration.java

```diff
@@ -18,8 +18,10 @@
  */
 package org.apache.polaris.persistence.relational.jdbc;
 
+import io.smallrye.config.ConfigMapping;
 import java.util.Optional;
 
+@ConfigMapping(prefix = "polaris.persistence.relational.jdbc")
```

Comment:
I do not oppose this change, but I'd like to highlight that this will impose the config naming convention defined in this annotation on downstream projects. It may be possible to provide a different config object for JDBC downstream or it may not be... I cannot say for sure.

Old code delegated runtime configuration to the server runtime, so this may be a behaviour change in downstream projects.

Given that NoSQL code follows a similar approach (not splitting interfaces and `@ConfigMapping`) this change looks reasonable to me.